repo_autoindex/_impl/kickstart.py

```diff
@@ -0,0 +1,235 @@
+        LOG.debug("repomd.xml: %s", self.entry_point_content)
+        all_entries.extend(await super()._repodata_entries())
+        all_entries.extend(await super()._package_entries())
+
+        ctx = TemplateContext()
+        nodes = [treeify(all_entries, index_href_suffix=index_href_suffix)]
+        while nodes:
+            node = nodes.pop()
+            yield GeneratedIndex(
+                content=ctx.render_index(index_entries=node.entries),
+                relative_dir=node.relative_dir,
+            )
+            nodes.extend(node.children)
+
+    async def _treeinfo_entries(self) -> list[IndexEntry]:
+        """
+        A treeinfo file might look like this:
+
+        [checksums]
+        images/boot.iso = sha256:f6be6ec48a4a610e25d591dcf98e1777c4274ed58c583fa64d0aea5b3ecffb18
+        images/efiboot.img = sha256:94d5500c4ba266ce77b06aa955d9041eea22129737badc6af56c283dcaec1c29
+        images/install.img = sha256:46171146377610cfa0deae157bbcc4ea146b3995c9b0c58d9f261ce404468abe
+        images/pxeboot/initrd.img = sha256:e0cd3966097c175d3aaf406a7f8c094374c69504c7be8f08d8084ab9a8812796
+        images/pxeboot/vmlinuz = sha256:370db9a3943d4f46dc079dbaeb7e0cc3910dca069f7eede66d3d7d0d5177f684
+
+        [general]
+        ; WARNING.0 = This section provides compatibility with pre-productmd treeinfos.
+        ; WARNING.1 = Read productmd documentation for details about new format.
+        arch = x86_64
+        family = Red Hat Enterprise Linux
+        name = Red Hat Enterprise Linux 8.0.0
+        packagedir = Packages
+        platforms = x86_64,xen
+        repository = .
+        timestamp = 1554367044
+        variant = BaseOS
+        variants = BaseOS
+        version = 8.0.0
+
+        [header]
+        type = productmd.treeinfo
+        version = 1.2
+
+        [images-x86_64]
+        boot.iso = images/boot.iso
+        efiboot.img = images/efiboot.img
+        initrd = images/pxeboot/initrd.img
+        kernel = images/pxeboot/vmlinuz
+
+        [images-xen]
+        initrd = images/pxeboot/initrd.img
+        kernel = images/pxeboot/vmlinuz
+
+        [release]
+        name = Red Hat Enterprise Linux
+        short = RHEL
+        version = 8.0.0
+
+        [stage2]
+        mainimage = images/install.img
+
+        [tree]
+        arch = x86_64
+        build_timestamp = 1554367044
+        platforms = x86_64,xen
+        variants = BaseOS
+
+        [variant-BaseOS]
+        id = BaseOS
+        name = BaseOS
+        packages = Packages
+        repository = .
+        type = variant
+        uid = BaseOS
+        """
+        out: list[IndexEntry] = [
+            IndexEntry(
+                href="treeinfo",
+                text="treeinfo",
+                size=str(len(self.treeinfo_content)),
+            ),
+        ]
+
+        treeinfo = configparser.ConfigParser()
+        treeinfo.read_string(self.treeinfo_content)
+
+        # Only attempt to parse a supported treeinfo format
```

Comment:
Not sure I'm in favor of this check, unless you know it fixes a specific problem.

It seems likely that other versions (all versions?) of the format work just fine.

If the tools were being updated so that a new version would be used, I think there's pretty high risk that this info would not reach anyone who realizes that this code needs to be updated.  Then it'd stop working, even if the updated format is totally backwards-compatible.  It might take a long time before anyone realizes the problem and figures out where to file an issue.

Could we just try to use the values no matter what?  If you're worried about it, the code could be written to catch errors and convert them into warnings here.

Or, a compromise would be to instead check that the version starts with "1.".  If the tools are maintained well, nobody should be making backwards-incompatible changes without first bumping the version to 2.x.

Reply:
I was hesitant to add this check anyway, so I'll remove it.